lib/pharos/phase.rb

```diff
@@ -61,5 +61,15 @@ def exec_script(script, vars = {})
     def parse_resource_file(path, vars = {})
       Pharos::YamlFile.new(resource_path(path)).read(vars)
     end
+
+    # @return [Hash]
+    def cluster_context
+      self.class.cluster_context
```

Comment:
```ruby
def cluster_context
  Pharos::Phase.cluster_context
end

def self.cluster_context
  @cluster_context ||= {}
end
```

On instance level this would have the same effect but without the class variable. `Phases::FooFoo.cluster_context` would get messed up though.

I suppose this ultimately belongs to `ClusterManager` which should pass a self-reference to all phases, sub-managers and whatever it creates.

Reply:
Yeah, the context could be a `Pharos::Phase#initialize` parameter like `config` if you wanted to make this fancier.